diff --git a/mobile/android/geckoview/src/main/java/org/mozilla/geckoview/GeckoRuntime.java b/mobile/android/geckoview/src/main/java/org/mozilla/geckoview/GeckoRuntime.java
--- a/mobile/android/geckoview/src/main/java/org/mozilla/geckoview/GeckoRuntime.java
+++ b/mobile/android/geckoview/src/main/java/org/mozilla/geckoview/GeckoRuntime.java
@@ -182,9 +182,10 @@ public final class GeckoRuntime implemen
         mAutocompleteStorageProxy = new Autocomplete.StorageProxy();
         mProfilerController = new ProfilerController();
 
-        if (sRuntime != null) {
-            throw new IllegalStateException("Only one GeckoRuntime instance is allowed");
-        }
+//      boerpatch: overly alarmistic when we've actually killed the runtime
+//        if (sRuntime != null) {
+//            throw new IllegalStateException("Only one GeckoRuntime instance is allowed");
+//        }
         sRuntime = this;
     }
 
@@ -398,12 +399,14 @@ public final class GeckoRuntime implemen
 
         if (!GeckoThread.init(info)) {
             Log.w(LOGTAG, "init failed (could not initiate GeckoThread)");
-            return false;
+            // boerpatch: overly pessimistic when we've actually killed the runtime
+            // return false;
         }
 
         if (!GeckoThread.launch()) {
             Log.w(LOGTAG, "init failed (GeckoThread already launched)");
-            return false;
+            // boerpatch: overly pessimistic when we've actually killed the runtime
+            // return false;
         }
 
         mSettings = settings;
@@ -516,7 +519,8 @@ public final class GeckoRuntime implemen
         runtime.attachTo(context);
 
         if (!runtime.init(context, settings)) {
-            throw new IllegalStateException("Failed to initialize GeckoRuntime");
+          // boerpatch: overly alarmistic when we've actually killed the runtime
+          // throw new IllegalStateException("Failed to initialize GeckoRuntime");
         }
 
         return runtime;
